### PR TITLE
RecoDecay: Improve azimuth calculations.

### PR DESCRIPTION
**Pull request by vkucera**

### CODE REVIEW

File: Analysis/Core/include/AnalysisCore/RecoDecay.h [hunk 1]
```diff
@@ -182,24 +182,43 @@ class RecoDecay
...
-  static double Phi(double px, double py)
+  /// Calculates azimuth from x and y components.
+  /// \param x,y  {x, y} components
+  /// \return azimuth within [0, 2π]
+  template <typename T, typename U>
+  static double Phi(T x, U y)
   {
-    // phi = pi+TMath::Atan2(-py,-px)
-    return (double)(o2::constants::math::PI + std::atan2(-py, -px));
+    // conversion from [-π, +π] returned by atan2 to [0, 2π]
```
Comment (ktf) [inline Analysis/Core/include/AnalysisCore/RecoDecay.h:191]: Please, no special characters....

Comment (vkucera) [reply Analysis/Core/include/AnalysisCore/RecoDecay.h:191]: You mean the "π"? It should be a Unicode character. Is it not?

Comment (ktf) [reply Analysis/Core/include/AnalysisCore/RecoDecay.h:191]: Exactly...

Comment (vkucera) [reply Analysis/Core/include/AnalysisCore/RecoDecay.h:191]: Use of Unicode characters in comments is explicitly encouraged by the [Coding Guidelines](https://rawgit.com/AliceO2Group/CodingGuidelines/master/comments_guidelines.html?showone=Special_Characters#Special_Characters).
What kind of problem do you see with Unicode?

**Comment by vkucera** [comment]
@jgrosseo @iarsene Any objections against merging?

**Comment by jgrosseo** [comment]
I don't, but I want to know if @ktf agrees because he made the comment

**Comment by jgrosseo** [comment]
btw we use mathutils in the DataModel. Is that better than std?

**Comment by ktf** [comment]
I just find unicode more difficult to parse, especially if in the end the code uses M_PI not π. Anyway, if it's in the Guidelines, it's a battle I will not fight... As long as we don't start having emojiis as variables...

**Pull request merged**